linker-interface/shared/src/main/scala/org/scalajs/linker/interface/StandardConfig.scala

```diff
@@ -80,6 +82,7 @@ final class StandardConfig private (
         sourceMap = true,
         relativizeSourceMapBase = None,
         outputPatterns = OutputPatterns.Defaults,
+        minify = false,
```

Comment:
Not that unlikely: I already a follow-up commit that minifies something non-names at https://github.com/scala-js/scala-js/pull/4931/commits/bceb8967bcfc2bac7652aa8335c909b980a5e535

Reply:
Yes, I saw that. I actually was wondering why we need to do that under `minify` and not always?

Reply:
I was going to ask about your opinion on that when we got there. We could do it always, but I fear the consequences on debugging-readability of the normally emitted code. For example it prevents to Ctrl+F for `$c_the_class_Name.prototype.methodName` to find the actual implementation of a method in a given class. All we would see is `$p.methodName`, but that would be the same for all the classes in the world, so it would much harder to get to the correct method.

Reply:
Hmm.. Yeah, that's a fair point I guess.

Somewhat independent of what we decide for the prototype thing, I think it makes sense to specify what `minify` means. Maybe something like: "Reduce code size at the cost of readability but not at the cost of performance (with a focus on what general purpose minifiers cannot do, but that is explicitly not part of the specification)"?

WDYT?

I'm also thinking if we need to factor "optimisation profiles" into the discussion, things like:
- DEBUG: Optimize output for step-by-step debugging
- RELOAD: Optimize output for fast hot reloading
- FAST: Optimize output for speed
- SMALL: Optimize output for size.

Then again, maybe profiles like this are better suited as a "porcelain" on top, adjusting the lower-level settings for to achieve this.

Reply:
> Somewhat independent of what we decide for the prototype thing, I think it makes sense to specify what `minify` means. Maybe something like: "Reduce code size at the cost of readability but not at the cost of performance (with a focus on what general purpose minifiers cannot do, but that is explicitly not part of the specification)"?
> 
> WDYT?

Yes, indeed. That spec matches my design. I'll add a comment.

Usually reducing size does not (negatively) affect speed.

One possible counter-example would be to factor out getter and setter functions. IIUC GCC does this. Despite inlining, there remains hundreds of `$p.foo = function() { return this.bar; }` and `$.foo = function(v) { this.bar = v; }` lying around. It's possible to factor them in with a pair of helpers:
```js
function $makeGetter(fieldName) {
  return (function() { return this[fieldName]; });
}
function $makeSetter(fieldName) {
  return (function(v) { this[fieldName] = v; });
}
```
Those can then be called as `$p.foo = $makeGetter("bar")`, which can become very short after name-minification of `$makeGetter`.

We would have to actually benchmark the resulting code if we attempted that, to make sure it doesn't negatively affect performance.